stripe_payments.php

```diff
@@ -1223,63 +1239,56 @@ public function storeAch(array $account_info, array $contact, $client_reference_
-            // Get bank account
-            $account = $this->handleApiRequest(
-                ['Stripe\Customer', 'retrieveSource'],
-                [($customer->id ?? $client_reference_id), $account_info['reference_id']],
-                $this->base_url . 'customers - retrieveSource'
-            );
+        // Add a new bank account to the same client
+        $account_data = $this->storeAch($account_info, $contact, $client_reference_id);
 
-            if (!empty($account['error'])) {
-                // Ignore any errors caused by attempting to fetch the old account
-                $this->Input->setErrors([]);
-            }
+        if ($this->Input->errors()) {
+            return false;
+        }
 
-            // Verify bank account
-            if (isset($account->customer) && $account->customer == $client_reference_id) {
-                try {
-                    $account->verify(['amounts' => [($_POST['first_deposit'] ?? 0), ($_POST['second_deposit'] ?? 0)]]);
-                } catch (Throwable $e) {
-                    $this->Input->setErrors(['verify' => ['error' => $e->getMessage()]]);
-                }
+        // Remove the old payment account if possible
+        if (false === $this->removeAch($client_reference_id, $account_reference_id)) {
+            // Ignore any errors caused by attempting to remove the old account
+            $this->Input->setErrors([]);
+        }
 
-                if ($this->Input->errors()) {
-                    return false;
-                }
-            }
+        return $account_data;
+    }
 
-            // Get current account
-            Loader::loadModels($this, ['GatewayManager']);
-            Loader::loadComponents($this, ['Record']);
-            $stored_account = $this->Record->select()
-                ->from('accounts_ach')
-                ->where('reference_id', '=', $account_reference_id)
-                ->fetch();
-            $stored_account->last4 = isset($stored_account->last4) ? $this->GatewayManager->systemDecrypt($stored_account->last4) : null;
-
-            return [
-                'last4' => $stored_account->last4,
-                'client_reference_id' => $client_reference_id,
-                'reference_id' => $account_reference_id
-            ];
-        } else {
-            // Add a new bank account to the same client
-            $account_data = $this->storeAch($account_info, $contact, $client_reference_id);
+    /**
+     * {@inheritdoc}
+     */
+    public function verifyAch(array $vars, $client_reference_id = null, $account_reference_id = null)
+    {
+        // Get bank account
+        $account = $this->handleApiRequest(
+            ['Stripe\Customer', 'retrieveSource'],
+            [$client_reference_id, $account_reference_id],
+            $this->base_url . 'customers - retrieveSource'
+        );
 
-            if ($this->Input->errors()) {
-                return false;
-            }
+        if (!empty($account['error'])) {
```

Comment:
I think $account is an object, but also I don't think we want to ignore errors in fetching the account.

Reply:
$account is an object only on a successful response, when an error occurs, an array is returned.